File: src/go/stack.go
package main

import "fmt"

// A simple stack using generics
type Stack[T any] struct {
	items []T
}

func (stack *Stack[T]) Push(value T) {
	stack.items = append(stack.items, value)
}

func (stack *Stack[T]) Pop() T {
	n := len(stack.items)
	if n <= 0 {
		panic("Cannot pop an empty stack!")
	}
	value := stack.items[n-1]
	stack.items = stack.items[:n-1]
	return value
}

func (stack *Stack[T]) Show() {
	fmt.Printf("%v\n", stack.items)
}

func main() {
	stack := Stack[int]{}
	stack.Push(1)
	stack.Push(2)
	stack.Push(3)
	stack.Push(4)
	stack.Pop()
	fmt.Printf("Stack: ")
	stack.Show()
}

File: src/javascript/Stack.js
class Stack {
    constructor() {
        this.stack = [];
    }

    push(value) {
        this.stack.push(value);
    }

    pop() {
        this.stack.pop();
    }

    read() {
        console.log(this.stack);
    }

    clear() {
        this.stack = [];
    }
}

function main() {
  let stack = new Stack();

  stack.push(1);
  stack.push(2);
  stack.push(3);
  stack.read();

  stack.pop();
  stack.read();

  stack.clear();
  stack.read();
}

main();
